src/components/bottom/ConnectedLiveGraphButtons.svelte

```diff
@@ -11,13 +11,26 @@
   import StandardButton from '../StandardButton.svelte';
   import Microbits from '../../script/microbit-interfacing/Microbits';
   import { startConnectionProcess } from '../../script/stores/connectDialogStore';
+  import { DeviceRequestStates } from '../../script/microbit-interfacing/MicrobitConnection';
 
   const handleInputDisconnectClick = () => {
-    Microbits.expelInputAndOutput();
+    Microbits.disconnect(DeviceRequestStates.INPUT);
   };
 
   const handleOutputDisconnectClick = () => {
-    Microbits.expelOutput();
+    Microbits.disconnect(DeviceRequestStates.OUTPUT);
+  };
+
+  const handleInputConnect = async () => {
+    if ($state.offerReconnect || $state.isInputAssigned) {
+      try {
+        await Microbits.reconnect(DeviceRequestStates.INPUT);
+      } catch (e) {
+        startConnectionProcess();
```

Comment:
Microbits.reconnect itself has some similar fallback. Perhaps we can remove it and make it the UI's responsibility to decide if a reconnect should start over from scratch with a dialog.

Reply:
Code is different here from ReconnectPrompt which has no fallback of its own. @microbit-robert please can you look and decide how you'd prefer it to work?

Reply:
I think both callers should be: 
```
  try {
    await Microbits.reconnect($state.reconnectState);
  } catch (e) {
    startConnectionProcess();
  } finally {
    stateOnStopOfferingReconnect();
  }
```

Then we can take `startConnectionProcess` and `stateOnStopOfferingReconnect` out of `Microbits.ts`.

Reply:
Made this change on this PR [b8e37ee](https://github.com/microbit-foundation/ml-trainer/pull/210/commits/b8e37ee08f1ccc770bbfe7b206c81d095dec3daf).